backend/battleship.py

```diff
@@ -0,0 +1,169 @@
+import argparse
+import random
+
+from web3 import Web3
+
+from constants import ABI
+from game import Game
+from log_setup import setup_logger
+
+logger = setup_logger()
+
+
+class BattleshipGame(Game):
+    board = 0
+
+    def __init__(self, rpc_url, contract_address):
+        self._rpc_url = rpc_url
+        self._contract_address = contract_address
+        self.board_size()
+        self.calculate_filled_cells()
+
+    def board_size(self, size=10):
+        self.board = [[0] * 10 for _ in range(size)]
+
+    def calculate_filled_cells(self):
+        """
+        Fill board with "ships".
+        """
+        for ship_size in [5, 4, 3, 3, 2]:
+            while True:
+                direction = random.choice(['horizontal', 'vertical'])
+                if direction == 'horizontal':
+                    row = random.randint(0, 9)
+                    col = random.randint(0, 10 - ship_size)  # Ship size
+                else:
+                    row = random.randint(0, 10 - ship_size)  # Ship size
+                    col = random.randint(0, 9)
+
+                # Check
+                if all(self.board[row + (direction == 'vertical') * i][col + (direction == 'horizontal') * i] == 0 for i in range(ship_size)):
+                    # Place ship and change cells
+                    for i in range(ship_size):
+                        self.board[row + (direction == 'vertical') * i][col + (direction == 'horizontal') * i] = 1
+                    break
+
+    @property
+    def timeout(self):
+        return self.timeout
+
+    @timeout.setter
+    def timeout(self, num_in_sec):
+        self.timeout = num_in_sec
+
+    def game_move_price(self, quantity):
+        self.game_move_price = quantity
+
+    @property
+    def reward_balance(self):
```

Comment:
Reward balance is stored in the on-chain part of every game, the function of these off-chain counterparts is to provide only the secret information when it's being explored and revealed by user moves.

Reply:
How this should looks like?